### PR TITLE
EVG-16073: Add an end point to get recent running versions of a task

### PR DESCRIPTION
**Pull request by hadjri**
[EVG-16073](https://jira.mongodb.org/browse/EVG-16073)

### Description 
New route was created allowing users to get recent versions of a task for a specific project. Allows for filtering on build variant, starting revision number, and limit on the number of results returned.
### Testing 
Added new unit tests for the DB function and rest route.

### CODE REVIEW

File: model/project_ref.go
Comment (malikchaya2) [inline model/project_ref.go:1065]: Can we elaborate on this comment, explaining what the function does?

File: model/version.go
Comment (malikchaya2) [inline model/version.go:602]: why was this removed?

**Review by malikchaya2 — CHANGES_REQUESTED** [review]

File: model/version.go
Comment (malikchaya2) [reply model/version.go:602]: waiting on this question

Comment (bynn) [reply model/version.go:602]: i think because we would be returning the same thing if we return in this if statement or out of it there's no point in doing the check

Comment (malikchaya2) [reply model/version.go:602]: oh, true! thanks!

File: rest/data/project.go
Comment (ablack12) [inline rest/data/project.go:109]: need a comment here

File: rest/route/project.go
Comment (ablack12) [inline rest/route/project.go:1014]: Preference for this to just error if h.opts.NumVersion < 0, and keep the default handling in GetProjectTasksWithOptions.

File: rest/route/project.go
Comment (ablack12) [inline rest/route/project.go:1030]: You can just return gimlet.NewJSONResponse(versions)

File: model/project_ref.go
Comment (ablack12) [inline model/project_ref.go:1067]: This doesn't specifically filter tasks that have _run_. A big reason cloud wants this is to filter out non-activated / scheduled tasks so that they don't have to paginate. We should also reflect this in the test.

**Review by ablack12 — CHANGES_REQUESTED** [review]

File: rest/route/project.go
Comment (ablack12) [inline rest/route/project.go:1010]: Can this be renamed to limit? I think that may be more in line with our existing routes.

**Review by ablack12 — APPROVED** [review]
LGTM with comment and assuming this is tested in staging before merging (and added to wiki).

**Comment by hadjri** [comment]
> LGTM with comment and assuming this is tested in staging before merging (and added to wiki).

Documentation: https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#get-tasks-for-a-project

Attached screenshot testing in staging with opts:
Screen Shot 2022-04-13 at 1 10 04 PM

**Comment by hadjri** [comment]
> LGTM with comment and assuming this is tested in staging before merging (and added to wiki).

Documentation: https://github.com/evergreen-ci/evergreen/wiki/REST-V2-Usage#get-tasks-for-a-project

Attached screenshot testing in staging with opts:
Screen Shot 2022-04-13 at 1 10 04 PM

**Pull request merged**
